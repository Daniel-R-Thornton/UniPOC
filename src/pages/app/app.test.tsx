import App from "./app";
import { render } from "@testing-library/react";
import { describe, test, expect } from "vitest";

describe("App test", () => {
  test("Should render when called", () => {
    const { getByRole } = render(App());
    expect(getByRole("main")).toBeTruthy();
  });

  test("Should match snapshot", () => {
    const { asFragment } = render(App());
    expect(asFragment()).toMatchSnapshot();
  });
});
