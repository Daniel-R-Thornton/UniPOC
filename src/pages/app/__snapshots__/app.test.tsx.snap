// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App test > Should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <header
      class="flex justify-between items-center px-8 py-2 bg-gray-200 h-16 "
    >
      <div
        class="logo"
      >
        Logo
      </div>
      <div
        class="login-button"
      >
        <button
          class="amplify-button amplify-field-group__control"
          type="button"
        >
          Login
        </button>
      </div>
    </header>
  </header>
  <main>
    <div>
      <article
        class="grid grid-cols-2 flex-col px-10 m-10 gap-2"
      >
        <figure
          class="w-full col-span-2"
        >
          <img
            alt="https://stlouis.citizenlab.co/uploads/e0d13d54-d706-4d53-af85-2f6c48c4c2fe/project/header_bg/1c7aa1cb-7c75-4293-8f9b-37fa72a73657/large_afd08489-8a69-4e30-9591-974d82716d46.jpeg"
            class="w-full h-30 object-fill aspect-[4/1]"
            src="https://stlouis.citizenlab.co/uploads/e0d13d54-d706-4d53-af85-2f6c48c4c2fe/project/header_bg/1c7aa1cb-7c75-4293-8f9b-37fa72a73657/large_afd08489-8a69-4e30-9591-974d82716d46.jpeg"
          />
          <figcaption
            class="hidden"
          />
        </figure>
        <section
          class="p-4 flex flex-col gap-2  col-span-2 lg:col-span-1"
        >
          <h2
            class="text-lg"
          >
            Ideas Portal 
          </h2>
          <p
            class=""
          >
            Are you ready to become an Idea Champion? Share your innovative ideas or build upon the ideas of others! This portal will remain open for the next year and we will refer to it throughout the engagement process. Participants whose ideas best address the challenges that our community faces will be invited to partner with us to further develop their ideas! We encourage one idea per card and building off the ideas of others. Only use the location feature if your idea is specific to a place. Remember, you can submit as many ideas as you want!
          </p>
        </section>
        <section
          class="flex flex-col gap-4 border  outline-slate-400 outline outline-1 bg-gray-200  col-span-2 lg:col-span-1"
        >
          <div
            class="px-10 m-10 "
          >
            <h2
              class="text-xl"
            >
              Welcome to the Ideas Portal
            </h2>
            <p
              class="text-sm pt-1"
            >
              This is a place to share your innovative ideas or build upon the ideas of others! This portal will remain open for the next year and we will refer to it throughout the engagement process. Participants whose ideas best address the challenges that our community faces will be invited to partner with us to further develop their ideas! We encourage one idea per card and building off the ideas of others. Only use the location feature if your idea is specific to a place. Remember, you can submit as many ideas as you want!
            </p>
          </div>
        </section>
      </article>
    </div>
    <div
      class="flex justify-between items-center bg-[#033d8b] p-8"
    >
      <h2
        class="text-2xl font-bold text-white"
      >
        Share your innovative ideas!
      </h2>
      <div
        class="flex justify-end"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          Submit an Idea
        </button>
      </div>
    </div>
    <div
      class="grid grid-cols-3  p-10  gap-3"
    />
  </main>
  <footer />
</DocumentFragment>
`;
